pkg/client/keycloak/gocloak_client_interface.go

```diff
@@ -0,0 +1,9 @@
+package keycloak
+
+import "github.com/Nerzal/gocloak/v11"
+
+// We create a type alias to gocloak.GoCloak to be able to autogenerate
+// mocks for it
+//
+//go:generate moq -out gocloak_client_interface_moq.go . goCloakClientInterface
+type goCloakClientInterface = gocloak.GoCloak
```

Comment:
The name `goCloakClientInterface` instead of `goCloakClient` is intentional.
The reason is that unfortunately the `github.com/Nerzal/gocloak/v11` package has two types `gocloak.Gocloak` and `gocloak.Client`, nd `gocloak.Client` is also used in other parts of our code. To differentiate it from `gocloak.Client` `goCloakClientInterface` has been decided as the name.